### PR TITLE
Image Submission

### PR DESCRIPTION
**Pull request by mattwebbio**
Adds functionality and tests for uploading and displaying images on the site.

### CODE REVIEW

**Comment by mattwebbio** [comment]
[Issue 2: Save Image Link](https://trello.com/c/X4BZOFzX/216-issue-2-save-image-link)

File: app/controllers/images_controller.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+class ImagesController < ApplicationController
+  def new
+    @image = Image.new
+  end
+
+  def create
```
Comment (darreneng) [inline app/controllers/images_controller.rb]: let's add a `show` action in this controller

File: app/controllers/images_controller.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    @image = Image.new(params.require(:image).permit(:url))
+
+    if @image.save
+      flash.now[:success] = 'Image uploaded successfully!'
```
Comment (darreneng) [inline app/controllers/images_controller.rb]: we'll want to redirect to our show action here

File: app/views/application/home.html.erb [hunk 3]
```diff
@@ -1 +1,17 @@
-<h1>Hello World</h1>
+<table class="table table-bordered">
```
Comment (darreneng) [inline app/views/application/home.html.erb]: we can move this into a separate branch for issue 3

File: test/controllers/images_controller_test.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    assert_response :success
+  end
+
+  test '"new" page has upload form' do
+    get '/images/new'
+    assert_select 'form'
+  end
+
+  test 'adding bad image URL fails' do
+    post '/images', params: { image: { url: 'iuashdfiuh' } }
```
Comment (darreneng) [inline test/controllers/images_controller_test.rb]: let's use `assert_difference` here

File: app/models/image.rb [hunk 5]
```diff
@@ -1,3 +1,3 @@
 class Image < ApplicationRecord
-  validates :url, format: { with: URI.regexp }
+  validates :url, format: { with: URI::DEFAULT_PARSER.make_regexp }
```
Comment (darreneng) [inline app/models/image.rb]: let's commit this with the original change

File: app/views/layouts/application.html.erb [hunk 6]
```diff
@@ -11,16 +11,32 @@
...
+        <%= yield %>
+      </div>
+    </div>
 
     <!-- Optional JavaScript -->
     <!-- jQuery first, then Popper.js, then Bootstrap JS -->
     <script src="https://code.jquery.com/jquery-3.2.1.slim.min.js" integrity="sha384-KJ3o2DKtIkvYIK3UENzmM7KCkRr/rE9/Qpg6aAZGJwFDMVNA/GpGFF93hXpG5KkN" crossorigin="anonymous"></script>
     <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.12.3/umd/popper.min.js" integrity="sha384-vFJXuSJphROIrBnz7yo7oB41mKfc8JzQZiCq4NCceLEaO4IHwicKwpJf9c9IpFgh" crossorigin="anonymous"></script>
-    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-beta.2/js/bootstrap.min.js" integrity="sha384-alpBpkh1PFOepccYVYDB4do5UnbKysX5WZXm3XxPqe5iKTfUKjNkCk9SaVuEZflJ" crossorigin="anonymous"></script>
+    <script src="https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/js/bootstrap.min.js" integrity="sha384-JjSmVgyd0p3pXB1rRibZUAYoIIy6OrQ6VrjIEaFf/nJGzIxFDsf4x0xIM+B07jRM" crossorigin="anonymous"></script>
```
Comment (darreneng) [inline app/views/layouts/application.html.erb]: let's commit this separately

**Pull request merged**